src/main/java/frc/robot/subsystems/swerve/swervestatehelpers/SwerveStateHelper.java

```diff
@@ -0,0 +1,115 @@
+
+public class SwerveStateHelper {
+
+    private final Supplier<Pose2d> robotPoseSupplier;
+    private final Swerve swerve;
+    private final SwerveConstants swerveConstants;
+    private final Supplier<Optional<Translation2d>> noteTranslationSupplier;
+
+    public SwerveStateHelper(Supplier<Pose2d> robotPoseSupplier, Supplier<Optional<Translation2d>> noteTranslationSupplier,
+            Swerve swerve) {
+        this.robotPoseSupplier = robotPoseSupplier;
+        this.swerve = swerve;
+        this.swerveConstants = swerve.getConstants();
+        this.noteTranslationSupplier = noteTranslationSupplier;
+    }
+
+    public ChassisSpeeds applyAimAssistOnInputsSpeeds(AimAssist aimAssistState, ChassisSpeeds inputSpeeds) {
+        return switch (aimAssistState) {
+            case SPEAKER -> getRotationAssistedSpeeds(
+                    inputSpeeds,
+                    swerve.getRobotRelativeVelocity(),
+                    () -> robotPoseSupplier.get().getRotation(),
+                    () -> SwerveMath.getRelativeTranslation(robotPoseSupplier.get().getTranslation(),new Translation2d(0,0)).getAngle(),
+                    swerveConstants
+            );
+            case AMP -> getRotationAssistedSpeeds(
+                    inputSpeeds,
+                    swerve.getRobotRelativeVelocity(),
+                    () -> robotPoseSupplier.get().getRotation(),
+                    () -> Rotation2d.fromDegrees(90),
+                    swerveConstants
+            );
+            case NOTE -> handleNoteAssistState(
+                    inputSpeeds,
+                    robotPoseSupplier,
+                    noteTranslationSupplier
+            );
+            case NONE -> inputSpeeds;
+        };
+    }
+
+
+    private ChassisSpeeds handleNoteAssistState(ChassisSpeeds inputSpeeds,
+            Supplier<Pose2d> robotPoseSupplier,
+            Supplier<Optional<Translation2d>> noteTranslationSupplier) {
+        if (noteTranslationSupplier.get().isEmpty()) {
+            return inputSpeeds;
+        }
+        Translation2d noteRelativeToRobot = SwerveMath.getPoseRelativeTranslation(robotPoseSupplier.get(),
+                noteTranslationSupplier.get().get());
+
+
+        double wantedHorizontalVelocity = swerveConstants.yMetersPIDController().calculate(0, noteRelativeToRobot.getY())
+                + inputSpeeds.vyMetersPerSecond;
+
+        return ChassisSpeeds.fromRobotRelativeSpeeds(
+                inputSpeeds.vxMetersPerSecond,
+                wantedHorizontalVelocity,
+                inputSpeeds.omegaRadiansPerSecond,
+                robotPoseSupplier.get().getRotation()
+        );
+    }
+
+
+    public Translation2d getRotationAxis(RotateAxis rotationAxisState) {
+        return switch (rotationAxisState) {
+            case MIDDLE_OF_ROBOT -> new Translation2d();
+            case FRONT_LEFT_MODULE -> swerveConstants.LOCATIONS[ModuleUtils.ModuleName.FRONT_LEFT.getIndex()];
```

Comment:
```suggestion
            case FRONT_LEFT_MODULE -> swerveConstants.LOCATIONS[ModuleName.FRONT_LEFT.getIndex()];
```
you can import it if you prefer your choice

Reply:
nah, i'll pass, if in RotateAxis i call it than ill call it here

Reply:
just saying in rotate axis the problem was that an object came from nowhere here you do like double import its not nowhere its from ModuleName